### PR TITLE
Glupotka dev

### PR DESCRIPTION
**Pull request by Glupotka**
Hello

### CODE REVIEW

**Review by gibbsiara — COMMENTED** [review]
Hejka, zacznę od tego, że praca (tak samo jak poprzednia) czytelna i przyjemna do przejrzenia. Odpowiadając na pytanie przy uzupełnianiu danych, ten sposób jest poprawny (uzupełnianie zbioru testowego na podstawie danych z treningowego), bo nie powoduje wycieku danych, bo model dalej "nie zna" wartości z testowego + nie masz sytuacji (jakbyś uzupełniała test na podstawie test), że dane w test są zdegenerowane (na przykładzie wieku -  załóżmy, że w całym zbiorze mamy wartości 0 do 75, a w testowym trafia się grupa 20-30).

Po za tym na plus:
- modele (więcej niż było wymagane w treści),
- wykresy ważności cech i i wnioski z nich,
- wykresy dla hiperparametrów i wnioski z nich.

Na minus:
- troszkę (może tylko dla mnie) zabrakło pokazania wyniku modelu na zbiorze treningowym, bo to też może pokazać czy model nie overfittuje.

**Review by Kacpere0 — APPROVED** [review]
Czytelne i logiczne wnioski. Ciekawe podejście ze zbadaniem ważności cech i usunięciem cech nieistotnych. Według mnie brakuje podsumowania na końcu z rozstrzygnięciem który model jest najlepszy dla tego zbioru.

**Review by wojt157 — APPROVED** [review]
Praca bardzo dobrze wykonana, przejrzysta. Dobrze się to czyta. Podoba mi się podejście do prewencji wycieku danych. Faktycznie, brakuje trochę werdyktu na końcu.

**Comment by GregW04** [comment]
Twoje review wylądowało
GENERAL COMMENT:
Praca ma dobrą strukturę i będzie super jeżeli poprawisz Minusy. PR nie akceptuje, błędy choć drobne to były dość inwazyjne na dalszą część pracy. Oznacz mnie proszę jak zcommitujesz zmiany - to zrobię rereview. Jeżeli masz jakieś pytania to śmiało pytaj

NA PLUS:
- imputacja danych (odpowiadając na pytanie czy ok - tak jest poprawnie), jedynie brakuje ustawienia samego seeda (do reprodukowalności pomiędzy uruchomieniami)
- Baseline model
- GridSearchCV
- badanie SVM z użyciem wielu kerneli
- wykresy metryki do hiperaparmetrów fajne (ale chce zobaczyć dla pasującej metryki)

NA MINUS:
- Zły dobór scalera (użycie StandardScalera jest błędne w tym przypadku) -> odsyłam do wykładu 1
- sprawdzenie test splita i następnie tuning modelu (X_test i y_test używa się na samym końcu i potem nie dotyka się już modelu!!!) -> jeżeli chcesz robić takie eksperymenty polecam train-val-test split.
- zła metryka do oceny modelu, accuracy jest wrażliwe na niezbalansowane dane (polecam użyć np. f1-score)
- w drzewach badanie tylko głębokości (fajnie byłoby sprawdzić inne hiperparamtry jak np. szerokość)

WOLNE MYŚLI:
- GridSearch jest dość słabym sposobem hyperparameter tuningu (już lepszy jest RandomSearch). Ja personalnie polecam Optuna (warto poczytać jak działa).
- Brak głębszego badania potencjalnego overfittingu/underfittingu (np. poprzez podział na train-val-test split i krzywe uczenia)
- Tylko 3 modele (można było jeszcze zbadać LogisticRegression/ inne warianty modelów liniowych, Lasy Losowe, Drzewa wzmacniane gradientowo)

**Review by GregW04 — CHANGES_REQUESTED** [review]
zmiany w MINUSACH
oraz 
shoutout dla @RakPiotr za znalezienie tego błędu w innym review. Tutaj też to jest do poprawy: https://github.com/Solvro/ml-wakacyjne-wyzwanie-2026/pull/95#pullrequestreview-4984119687
